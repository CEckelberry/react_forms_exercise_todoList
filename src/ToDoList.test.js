import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import ToDoList from "./toDoList";

it("renders without crashing", () => {
    render(<ToDoList />);
});

it("matches the snapshot", () => {
    const { asFragment } = render(<ToDoList />);
    expect(asFragment()).toMatchSnapshot();
});

it("should add a new box", () => {
    const { queryByText, getByLabelText } = render(<ToDoList />);
    const taskInput = getByLabelText("Todo Task:");
    const btn = queryByText("Add Task");
    
    expect(queryByText('mow the lawn')).not.toBeInTheDocument();
    expect(queryByText('Todo Task:')).toBeInTheDocument();

    fireEvent.change(taskInput, { target: {value: 'mow the lawn'}});
    fireEvent.click(btn);

    expect(queryByText('mow the lawn')).toBeInTheDocument();
})