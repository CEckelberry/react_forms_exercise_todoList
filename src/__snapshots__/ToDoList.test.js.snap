// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      To Do List!
    </h1>
    <div>
      <form>
        <div>
          <label
            for="task"
          >
            Todo Task:
          </label>
          <input
            id="task"
            name="task"
            type="text"
            value=""
          />
          <button>
            Add Task
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
